src/index.ts

```diff
@@ -0,0 +1,32 @@
+import * as fs from 'fs';
+import * as yaml from 'js-yaml';
+import { ActionInput, VersionDoc, Tag } from './types';
+import { getActionInputs, getTagsPromises, updateVersions } from './helpers';
+
+async function run(): Promise<void> {
+
+    // Get the GH token and version file path
+    const inputs: ActionInput = getActionInputs([
+        { name: 'github_token', options: { required: true } },
+        { name: 'version_file_path', options: { required: true } },
+    ]);
+
+    // Get the repos to cycle through later
+    let doc: VersionDoc = yaml.load(fs.readFileSync(inputs.version_file_path, 'utf8'));
```

Comment:
This isn't validating the file has the correct syntax, just asserting to the compiler that it does. This can lead to awkward to identify bugs when the object turns out not to have the correct shape at runtime when it's used elsewhere in the code. There are libraries like [yup](https://www.npmjs.com/package/yup) and [zod](https://www.npmjs.com/package/zod) that provide ways to build validators that can be typechecked as validating what they say they do. To roll your own, Typescript has syntax for a boolean check that also validates a type:

```typescript
export const validateVersionDoc = (yaml): yaml is VersionDoc =>
    typeof yaml === 'object' &&
    typeof yaml.versions === 'object' &&
    Object.keys(yaml).length === 1 && // Versions is only property - catches yaml indentation errors
    Object.entries(yaml).every(([k, v]) => 
      typeof k === 'string' && typeof v === 'string'
      // Could also validate that the version string looks like a semantic version here
    );

// ...

let doc = yaml.load(fs.readFileSync(inputs.version_file_path, 'utf8'));
if(!validateVersionDoc(doc)) {
    throw new Error('Version file is not valid');
}
// The compiler now knows that doc has type VersionDoc
```
All that said - given this is a small code base, the bug won't happen far from the cause. So this is a nice-to-have rather than a blocker.